### PR TITLE
Improvement in Previous Attempts leaderboard.

### PR DESCRIPTION
**Pull request by thedevelopersanjeev**
Fixes part 3 of #181 

Changes: 1. Previous attempts leaderboard is now the same as on the website.
2. Replaced ListView with RecyclerView for better performance.

Screenshots for the change:
Screenshot_1558480227

### CODE REVIEW

File: app/src/main/java/com/codingblocks/cbonlineapp/adapters/QuizAttemptListAdapter.kt [hunk 1]
```diff
@@ -1,67 +1,57 @@
...
-    }
-
-    override fun getItem(i: Int): QuizAttempt? {
-        return list!![i]
+class QuizAttemptListAdapter(private val dataset: ArrayList<QuizAttempt>) :
+    RecyclerView.Adapter<QuizAttemptListAdapter.QuizAttemptViewHolder>() {
+
+    var onItemClick: ((QuizAttempt)->Unit) ?= null
+
+    inner class QuizAttemptViewHolder(val view: View) : RecyclerView.ViewHolder(view) {
```
Comment (aggarwalpulkit596) [inline app/src/main/java/com/codingblocks/cbonlineapp/adapters/QuizAttemptListAdapter.kt]: No need of inner class 
A simple class would be fine

File: app/src/main/java/com/codingblocks/cbonlineapp/adapters/QuizAttemptListAdapter.kt [hunk 2]
```diff
@@ -1,67 +1,57 @@
...
+        var numberTextView: TextView? = null
+        var dateTextView: TextView? = null
+        var scoreTextView: TextView? = null
+        var statusTextView: TextView? = null
+
+        init {
+            numberTextView = view.findViewById(R.id.numberTv)
+            dateTextView = view.findViewById(R.id.dateTv)
```
Comment (aggarwalpulkit596) [inline app/src/main/java/com/codingblocks/cbonlineapp/adapters/QuizAttemptListAdapter.kt]: Don't use this 
Synthetic data binding will work here as well

File: app/src/main/java/com/codingblocks/cbonlineapp/adapters/QuizAttemptListAdapter.kt [hunk 3]
```diff
@@ -1,67 +1,57 @@
...
-
-        }
-        view.setOnClickListener {
-            listener.onItemClick(position, e.id!!)
-        }
-
-        return view
+    // Replace the contents of view
+    override fun onBindViewHolder(holder: QuizAttemptViewHolder, position: Int) {
+        val currentAttempt = dataset[position]
```
Comment (aggarwalpulkit596) [inline app/src/main/java/com/codingblocks/cbonlineapp/adapters/QuizAttemptListAdapter.kt]: Do this work in QuizAttemptViewHolder

File: app/src/main/java/com/codingblocks/cbonlineapp/adapters/QuizAttemptListAdapter.kt (hunk 1 above)
Comment (thedevelopersanjeev) [reply app/src/main/java/com/codingblocks/cbonlineapp/adapters/QuizAttemptListAdapter.kt]: done.

File: app/src/main/java/com/codingblocks/cbonlineapp/adapters/QuizAttemptListAdapter.kt (hunk 3 above)
Comment (thedevelopersanjeev) [reply app/src/main/java/com/codingblocks/cbonlineapp/adapters/QuizAttemptListAdapter.kt]: done

Comment (thedevelopersanjeev) [reply app/src/main/java/com/codingblocks/cbonlineapp/adapters/QuizAttemptListAdapter.kt]: Yes, it works.

File: app/src/main/java/com/codingblocks/cbonlineapp/adapters/QuizAttemptListAdapter.kt (hunk 2 above)
Comment (thedevelopersanjeev) [reply app/src/main/java/com/codingblocks/cbonlineapp/adapters/QuizAttemptListAdapter.kt]: done.

**Comment by aggarwalpulkit596** [comment]
Claim 300 for this

**Pull request merged**